src/WebJobs.Script/Rpc/FunctionRegistration/FunctionDispatcher.cs

```diff
@@ -74,11 +73,11 @@ public void WorkerError(WorkerErrorEvent workerError)
                     CreateWorkerState,
                     (config, state) =>
                     {
-                        _erroredChannels.Add(state.Channel);
+                        erroredChannel.Dispose();
```

Comment:
@fabiocav @pragnagopa - this is a new change that calls `Dispose()` on WorkerErrorEvent. The `WorkerErrorEvent` method in LanguageWorkerChannel now just publishes a WorkerErrorEvent instead of handling some Dispose logic and publishing the WorkerErrorEvent.

This also makes it so that WorkerErrorEvents published by [FunctionRpcService](https://github.com/Azure/azure-functions-host/blob/a26dc28c05a2a18922a5116303979502088d9caa/src/WebJobs.Script/Rpc/FunctionRpcService.cs#L60) on write errors cause the same cleanup behavior.

See changes made to `HandleWorkerError` and `Dispose` in LanguageWorkerChannel as well. Thanks!

Reply:
LGTM!